src/engine.c

```diff
@@ -486,7 +533,26 @@ int draw_score(tetris_context_t *ctx) {
 
     SDL_RenderCopyF(ctx->renderer, text_texture, NULL, &dst);
 
-    return 0;
+	/* Draw the score value. */
+	snprintf(buffer, 20, "%d", ctx->score);
+	int i;
+	for (i = strlen(buffer); i < 20; ++i)
+		buffer[i] = ' ';
+	buffer[i] = 0;
+
+	surface = TTF_RenderText_Solid(ctx->font, buffer, color);
+	text_texture = SDL_CreateTextureFromSurface(ctx->renderer, surface);
```

Comment:
there's a memory leak here because it didn't free/destroy the previous ```surface``` and ```text_texture``` before reassigning it.  suggestion: 
```
	SDL_FreeSurface(surface);
	SDL_DestroyTexture(text_texture);
```
before line 536